pkg/controller/pxc/vs.go

```diff
@@ -4,18 +4,40 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+	"net/url"
 	"time"
 )
 
-func (vs VersionServiceClient) GetExactVersion(desiredVersion string, currentVersion string) (DepVersion, error) {
+func (vs VersionServiceClient) GetExactVersion(desiredVersion, currentVersion string, versionMeta currVersionMeta) (DepVersion, error) {
 	client := http.Client{
 		Timeout: 5 * time.Second,
 	}
-	url := fmt.Sprintf("%s/api/versions/v1/pxc/%s/%s?databaseVersion=%s", vs.URL, vs.OpVersion, desiredVersion, currentVersion)
-	req, err := http.NewRequest("GET", url, nil)
+
+	requestURL, err := url.Parse(fmt.Sprintf("%s/api/versions/v1/pxc/%s/%s", vs.URL, vs.OpVersion, desiredVersion))
```

Comment:
The whole point of using gRPC and Swagger is that you don't have to do _that_ – both API clients are generated from the specification. By not using them, you miss all the benefits like compile-time checks, runtime request validation, etc, and also just do more work – you spent time writing spec, and now you are not using it. Why?